### PR TITLE
Acceptance tests added for volume and snapshotRule resource

### PR DESCRIPTION
**Pull request by mkmayank-dell**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: powerstore/resource_snapshotrule.go [hunk 1]
```diff
@@ -71,8 +71,9 @@ func (r resourceSnapshotRuleType) GetSchema(_ context.Context) (tfsdk.Schema, di
 				Computed:            true,
 				Description:         "The time zone identifier for applying the time zone to the time_of_day for a snapshot rule.",
 				MarkdownDescription: "The time zone identifier for applying the time zone to the time_of_day for a snapshot rule.",
-				PlanModifiers: tfsdk.AttributePlanModifiers{
-					DefaultAttribute(types.String{Value: string(gopowerstore.TimeZoneEnumUTC)}),
+				PlanModifiers:       tfsdk.AttributePlanModifiers{
+					// DefaultAttribute(types.String{Value: string(gopowerstore.TimeZoneEnumUTC)}),
```
Comment (Sakshi-dell) [inline powerstore/resource_snapshotrule.go:75]: if it not required then please remove it.

File: powerstore/resource_snapshotrule_test.go [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+}
+
+// Test to Update SnapShotRule with invalid values
+func TestAccSnapshotRule_ReadSnapShotRuleUnavailable(t *testing.T) {
+	if os.Getenv("TF_ACC") == "" {
+		t.Skip("Dont run with units tests because it will try to create the context")
+	}
+
+	resource.Test(t, resource.TestCase{
+		// PreCheck:                 func() { testAccPreCheck(t) },
```
Comment (Sakshi-dell) [inline powerstore/resource_snapshotrule_test.go]: Any reason for disabling precheck?

File: powerstore/resource_snapshotrule.go (hunk 1 above)
Comment (mkmayank-dell) [reply powerstore/resource_snapshotrule.go:75]: what if in future reader wonder what could have been achieved by not using default attribute for this particular param but for others

so left a comment there in next line explaining the reason

File: powerstore/resource_snapshotrule_test.go (hunk 2 above)
Comment (mkmayank-dell) [reply powerstore/resource_snapshotrule_test.go]: ah its a typo, removing the comment

File: powerstore/provider_test.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"os"
+	"testing"
+)
+
+// var testAccProviders map[string]func() tfsdk.Provider
+var testProvider tfsdk.Provider
+var testProviderFactory map[string]func() (tfprotov6.ProviderServer, error)
+var endpoint = os.Getenv("ENDPOINT")
```
Comment (shenda1) [inline powerstore/provider_test.go]: The environment variable names look generic. Can we have specific to PowerStore e.g. POWERSTORE_ENDPOINT?

**Review by shenda1 — COMMENTED** [review]
Please attach the coverage report

File: powerstore/provider_test.go (hunk 3 above)
Comment (mkmayank-dell) [reply powerstore/provider_test.go]: updated

**Comment by mkmayank-dell** [comment]
coverage: 78.8% of statements
ok      terraform-provider-powerstore/powerstore        112.379s        coverage: 78.8% of statements


image

File: powerstore/resource_snapshotrule.go (hunk 1 above)
Comment (Sakshi-dell) [reply powerstore/resource_snapshotrule.go:75]: Then please mark it as TODO, we can not leave comments like this

Comment (mkmayank-dell) [reply powerstore/resource_snapshotrule.go:75]: comments should be there as many as we can right ?
its not a TODO, its there to tell why we not including it

File: powerstore/provider_test.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	if v := os.Getenv("POWERSTORE_ENDPOINT"); v == "" {
+		t.Fatal("POWERSTORE_ENDPOINT must be set for acceptance tests")
+	}
+
+}
+
+const EmptyEndpointConfig = `
+provider "powerstore" {
+	username = "username"
+	password = "password"
```
Comment (shenda1) [inline powerstore/provider_test.go]: endpoint is needed?

File: powerstore/resource_snapshotrule_test.go [hunk 5]
```diff
@@ -0,0 +1,448 @@
...
+}
+
+// Test to Create SnapShotRule with mutually exclusive params available
+func TestAccSnapshotRule_CreateSnapShotRuleWithTimeOfdayandInterval(t *testing.T) {
+	if os.Getenv("TF_ACC") == "" {
+		t.Skip("Dont run with units tests because it will try to create the context")
+	}
+
+	resource.Test(t, resource.TestCase{
+		// PreCheck:                 func() { testAccPreCheck(t) },
```
Comment (shenda1) [inline powerstore/resource_snapshotrule_test.go]: if this statement is not required, please remove it

Comment (mkmayank-dell) [reply powerstore/resource_snapshotrule_test.go]: its a typo, removing comment

File: powerstore/provider_test.go (hunk 4 above)
Comment (mkmayank-dell) [reply powerstore/provider_test.go]: nope, didn't use it anywhere,
removing

**Comment by mkmayank-dell** [comment]
PASS
coverage: 69.3% of statements
ok      terraform-provider-powerstore/powerstore        142.762s        coverage: 69.3% of statements

image

**Review by shenda1 — APPROVED** [review]

**Pull request merged**